### PR TITLE
Main

### PR DESCRIPTION
**Pull request by OlegMR0**

### CODE REVIEW

File: src/main/java/core/basesyntax/ArrayList.java [hunk 1]
```diff
@@ -1,48 +1,129 @@
...
+        }
+        System.arraycopy(toArray(list), 0, array, size(), list.size());
+        size += list.size();
     }
 
     @Override
     public T get(int index) {
-        return null;
+        if (index < 0 || index >= size) {
+            throw new ArrayListIndexOutOfBoundsException("");
```
Comment (uliana-andrushchuk) [inline src/main/java/core/basesyntax/ArrayList.java:61]: Please, add error message

File: src/main/java/core/basesyntax/ArrayList.java [hunk 2]
```diff
@@ -1,48 +1,129 @@
...
+        }
+        array[index] = value;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        if (index < 0 || index > size - 1) {
+            throw new ArrayListIndexOutOfBoundsException("");
+        }
```
Comment (uliana-andrushchuk) [inline src/main/java/core/basesyntax/ArrayList.java:78]: Please, extract this logic to separate method

File: src/main/java/core/basesyntax/ArrayList.java [hunk 3]
```diff
@@ -1,48 +1,129 @@
...
+    public ArrayList() {
+        array = (T[]) new Object[DEFAULT_ARRAY_SIZE];
+    }
+
+    public ArrayList(int size) {
+        if (size <= DEFAULT_ARRAY_SIZE) {
+            array = (T[]) new Object[DEFAULT_ARRAY_SIZE];
+        } else {
+            array = (T[]) new Object[size];
+        }
```
Comment (uliana-andrushchuk) [inline src/main/java/core/basesyntax/ArrayList.java:19]: You should have possibility to create ArrayList with capacity which is less than DEFAULT_ARRAY_SIZE

**Review by uliana-andrushchuk — CHANGES_REQUESTED** [review]

File: src/main/java/core/basesyntax/ArrayList.java [hunk 4]
```diff
@@ -1,48 +1,125 @@
...
+    }
+
+    private void checkIndex(int index) {
+        if (index < 0 || index >= size) {
+            throw new ArrayListIndexOutOfBoundsException("Wrong index");
+        }
     }
 }
+
+
```
Comment (kozhukhovsky) [inline src/main/java/core/basesyntax/ArrayList.java:125]: remove empty lines

**Review by kozhukhovsky — APPROVED** [review]

**Pull request closed without merging**